Integration tests start to fail on Travis

Example:

```
Traceback (most recent call last):
  File "/home/user/google-cloud-sdk/platform/google_appengine/dev_appserver.py", line 96, in <module>
    _run_file(__file__, globals())
  File "/home/user/google-cloud-sdk/platform/google_appengine/dev_appserver.py", line 90, in _run_file
    execfile(_PATHS.script_file(script_name), globals_)
  File "/home/user/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/devappserver2.py", line 617, in <module>
    main()
  File "/home/user/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/devappserver2.py", line 605, in main
    dev_server.start(options)
  File "/home/user/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/devappserver2.py", line 266, in start
    env_variables=parsed_env_variables)
  File "/home/user/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/application_configuration.py", line 957, in __init__
    env_variables)
  File "/home/user/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/application_configuration.py", line 140, in __init__
    self._config_path)
  File "/home/user/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/application_configuration.py", line 518, in _parse_configuration
    with open(configuration_path) as f:
IOError: [Errno 2] No such file or directory: '../webapp'
--- FAIL: TestFileResults (0.72s)
panic: unable to find webserver URL [recovered]
	panic: unable to find webserver URL
```

Looks like `dev_appserver` is having trouble finding the `webapp` directory (where we have our `app.yaml`). A few hypotheses come to my mind:

* A new version of `dev_appserver` no longer supports relative paths (not likely).
* CWD is wrong on Travis.

cc @lukebjerring

#616 worked around the failure I originally reported, but now we are seeing a new failure, again from gcloud SDK:

```
panic: Metadata fetch failed for 'instance/attributes/gae_backend_version': metadata server returned HTTP 404 [recovered]
	panic: Metadata fetch failed for 'instance/attributes/gae_backend_version': metadata server returned HTTP 404
```

--- a/webapp/about_handler.go
+++ b/webapp/about_handler.go
@@ -13,7 +13,12 @@ import (
 
 func aboutHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := appengine.NewContext(r)
-	version := strings.Split(appengine.VersionID(ctx), ".")[0]
+	var version string
+	if appengine.IsDevAppServer() {
+		version = strings.Split(appengine.VersionID(ctx), ".")[0]
+	} else {
+		version = "local-dev"
+	}
 	data := struct {
 		Version string
 	}{

